### ISSUE TITLE
Central text

### ISSUE THREAD

**Issue by zReport02** [KyoriPowered/adventure#655]
Is there a way to center the message text directly from minimessage?

**Comment by jpenilla** [comment]
No, ignoring the issue of how to measure width of components, how is minimessage to know the total width of where the component will be displayed?

**Comment by ryantheleach** [comment]
Speaking as someone who helped implement centered text for sponges help system, here be dragons.

Centering relies upon font metrics, and not just who the recipient is, but what their font is set to, (which last I looked wasn't always sent correctly even when you know the chat width settings)

Not to mention the countless places where adventure components end up used:

Signs, Books, Chat, Tab, Display Names, Scoreboards, Holograms, Console Output, Action Bar, Titles, Item Names, Lore, Tool Tips.

The only remotely feasible way to implement this in a sane manner, is if Mojang did it themselves.

**Comment by kezz** [comment]
Thanks for the suggestion, however this sounds like something that would be better suited for a community library: https://docs.adventure.kyori.net/community-libraries.html
